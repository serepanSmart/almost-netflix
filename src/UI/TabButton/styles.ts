import styled from 'styled-components';
import Colors from '../Theme/Colors';

export const FilterItem = styled.button<{ active?: boolean }>`
  position: relative;
  padding-bottom: 22px;
  border: none;
  background: transparent;
  font-size: 16px;
  text-transform: uppercase;
  font-weight: 600;
  z-index: 2;
  color: ${({ active }) => active
    ? Colors.Scarlet
    : Colors.White};
  cursor: pointer;
  border-bottom: 2px solid ${({ active }) => active
    ? Colors.Scarlet
    : 'transparent'};
  &:hover {
    color: ${Colors.Scarlet};
  }
  &:not(:last-child) {
    margin-right: 30px;
  }
`;