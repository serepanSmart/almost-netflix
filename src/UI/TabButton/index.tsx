import React, { useCallback, PropsWithChildren } from 'react';
import { FilterItem } from './styles';

export interface ITab {
  active: boolean;
  value: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  disabled?: boolean;
  id?: string;
}

const TabButton: React.FC<PropsWithChildren<ITab>> = ({
  active,
  onClick,
  disabled,
  value,
  id = value,
}) => {
  const handleClick = useCallback(
    (evt: React.MouseEvent<HTMLButtonElement>): void => {
      if (onClick) {
        onClick(evt);
      }
    },
    [onClick],
  );

  return (
    <FilterItem
      active={active}
      disabled={disabled}
      onClick={handleClick}
      value={value}
      type="button"
      id={id} // id here is value by default, but it can be changed
    >
      {value}
    </FilterItem>
  );
};

export default TabButton;
