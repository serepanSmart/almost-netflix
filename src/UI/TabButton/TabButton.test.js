import { render, fireEvent } from '@testing-library/react';
import TabButton from '.';

const handleClick = jest.fn();
const active = false;
const disabled = false;
const value = 'TabButton';
const id = 'id';

describe('TabButton component', () => {
  let component;
  beforeEach(() => {
    component = (props) =>
      render(
        <TabButton
          active={active}
          disabled={disabled}
          onClick={handleClick}
          value={value}
          type='button'
          id={id}
          {...props}
        />,
      );
  });

  describe('TabButton render tests', () => {
    test('should define TabButton component', () => {
      expect(component()).toBeDefined();
    });

    test('should be rendered with value prop', () => {
      const { getByText } = component();
      expect(getByText(value)).toBeInTheDocument();
    });

    test('should match snapshot with default props', () => {
      expect(component()).toMatchSnapshot();
    });

    test('should have text content', () => {
      const { getByText } = component({ disabled: true });
      expect(getByText(value)).toHaveTextContent(value);
    });

    test('should have "id" attr', () => {
      const { getByRole } = component();
      expect(getByRole('button')).toHaveAttribute('id');
    });

    test('should be disabled', () => {
      const { getByText } = component({ disabled: true });
      expect(getByText(value)).toBeDisabled();
    });

  });
  describe('TabButton event handlers', () => {
    test('should call onClick prop when clicked', () => {
      const { getByText } = component();
      fireEvent.click(getByText(value));
      expect(handleClick).toHaveBeenCalledTimes(1);
    });
  });
});
