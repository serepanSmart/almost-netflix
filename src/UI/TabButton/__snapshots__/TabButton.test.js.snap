// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton component TabButton render tests should match snapshot with default props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  padding-bottom: 22px;
  border: none;
  background: transparent;
  font-size: 16px;
  text-transform: uppercase;
  font-weight: 500;
  z-index: 2;
  color: #fff;
  cursor: pointer;
  border-bottom: 2px solid transparent;
}

.c0:hover {
  color: #f65261;
}

.c0:not(:last-child) {
  margin-right: 30px;
}

<body>
    <div>
      <button
        class="c0"
        id="id"
        type="button"
        value="TabButton"
      >
        TabButton
      </button>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  padding-bottom: 22px;
  border: none;
  background: transparent;
  font-size: 16px;
  text-transform: uppercase;
  font-weight: 500;
  z-index: 2;
  color: #fff;
  cursor: pointer;
  border-bottom: 2px solid transparent;
}

.c0:hover {
  color: #f65261;
}

.c0:not(:last-child) {
  margin-right: 30px;
}

<div>
    <button
      class="c0"
      id="id"
      type="button"
      value="TabButton"
    >
      TabButton
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
